src/rows.ts

```diff
@@ -126,6 +126,41 @@ export class Rows {
     return columns;
   }
 
+  /**
+   * Rows.toObjects
+   * 
+   * Call this if you need to ouput the rows as objects.
+   * 
+   * Will return an empty array if there are no entries in the table.
+   * 
+   *     const rows = db.query("SELECT name FROM users;").toObjects();
+   */
+  toObjects<T extends any = Record<string, any>>(): T[] {
```

Comment:
While I like the signature, I don't like the implementation. I'd prefer to return an iterator here, so that there is no big up-front read necessary. (If a user wants an array, they can use the spread operator: `[...db.query('SELECT * FROM people').asObjects()]`)

I was thinking of having a new class (e.g. `RowObjects`?, feel free to come up with a better name 😅 ), which stores the columns out put and basically consumes the underlying `Rows` iterator, transforming each row to an object as it passes through.

For that implementation I think I would also like to use `asObjects`, rather `toObjects`.

Reply:
I was just thinking about the try catch: You should be able to check `this === Empty` instead 😃

Reply:
Fair enough. The only thing I am not sure if I am too fond of is to create a new class to handle this. Wouldn't it be better to add an option argument to `query()`? If you strongly prefer to create a new class, I can do that, but I think it would keep the codebase cleaner if we add an argument to `query()`, add a private property to the class called `_columns` (or something) so that we only have to call `columns()` once,  and alter `next()` to facilitate this change. Let me know what you think! 😄

Reply:
I just pushed a commit where you can see what I mean above. If you prefer to use a class I can revert the commit

Reply:
I see your point, and I agree. I will change it as per the first option, but I might need some feedback in regards to the iterator (as this will be the first time I actually write one) 😄

Reply:
No worries, that’s what code reviews are for 😀

Reply:
So I implemented RowObjects 🎉 Let me know if there are any things to improve!